### PR TITLE
Add UT for list struct based executors

### LINKED ISSUE #5534: Add UT for list struct based executors
## Enhancement

While tidb currently uses tree struct base executor, older versions of tidb and tispark still use list struct base executors.

The difference is that the list struct base executors does not have an executor_id and the list struct base executors is held by DAGRequest.executors.
https://github.com/pingcap/tiflash/blob/0bafed9b2be3da85a9ec8cb3e160250ee67446ec/dbms/src/Flash/Coprocessor/DAGQueryBlock.cpp#L170-L232

Currently, ExecutorTestUtils only supports tree struct base executor, and we need to find a way to add UTs for list struct base executor.

related issue: https://github.com/pingcap/tiflash/issues/4609

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #5534

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TestUtils/tests/gtest_mock_executors.cpp [hunk 1]
```diff
@@ -340,5 +340,44 @@ try
...
+    }
+
+    {
+        auto request = context
+                           .scan("test_db", "test_table")
+                           .filter(eq(col("s1"), col("s2")))
+                           .aggregation(Max(col("s1")), col("s2"))
+                           .filter(eq(col("s2"), lit(Field("1", 1))))
+                           .topN("s2", false, 10)
+                           .buildToListStruct(context);
```
Comment (ywqzzy) [inline dbms/src/TestUtils/tests/gtest_mock_executors.cpp:371]:
```suggestion
                           .build(context, "list");
```
maybe better?

**Review by ywqzzy — COMMENTED** [review]
others LGTM

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by gengliqi — APPROVED** [review]
LGTM

**Pull request merged**